api/src/main/java/io/strimzi/api/kafka/model/CruiseControlSpec.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.fabric8.kubernetes.api.model.ResourceRequirements;
+import io.strimzi.api.kafka.model.template.CruiseControlTemplate;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.sundr.builder.annotations.Buildable;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "replicas", "image", "config",
+        "livenessProbe", "readinessProbe",
+        "jvmOptions", "resources",
+        "logging", "tlsSidecar", "template"})
+@EqualsAndHashCode
+public class CruiseControlSpec implements UnknownPropertyPreserving, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    public static final String FORBIDDEN_PREFIXES = "bootstrap.servers, zookeeper., ssl., security., failed.brokers.zk.path,"
+        + "webserver.http.port, webserver.http.address, webserver.api.urlprefix, metric.reporter.sampler.bootstrap.servers, metric.reporter.topic, metric.reporter.topic.pattern,"
+        + "partition.metric.sample.store.topic, broker.metric.sample.store.topic, capacity.config.file, skip.sample.store.topic.rack.awareness.check, cruise.control.metrics.topic"
+        + "cruise.control.metrics.reporter.bootstrap.servers, sasl. interceptor.classes";
+    public static final String FORBIDDEN_PREFIX_EXCEPTIONS = "";
+
+    private int replicas;
+    private String version;
+    private String image;
+    private TlsSidecar tlsSidecar;
+    private ResourceRequirements resources;
+    private Probe livenessProbe;
+    private Probe readinessProbe;
+    private JvmOptions jvmOptions;
+    private Logging logging;
+    private CruiseControlTemplate template;
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    private Map<String, Object> config = new HashMap<>(0);
+
+    @Description("The number of pods in the `Deployment`.")
+    @Minimum(1)
```

Comment:
Can we support `@Minimum(0)`? It might be useful as a way to pause CC.

Reply:
Yes, we can, I'll update it